pay-api/src/pay_api/services/non_sufficient_funds.py

```diff
@@ -0,0 +1,192 @@
+
+    @property
+    def id(self):
+        """Return the _id."""
+        return self._id
+
+    @id.setter
+    def id(self, value: int):
+        """Set the _id."""
+        self._id = value
+        self._dao.id = value
+
+    @property
+    def invoice_id(self):
+        """Return the _invoice_id."""
+        return self._invoice_id
+
+    @invoice_id.setter
+    def invoice_id(self, value: int):
+        """Set the _invoice_id."""
+        self._invoice_id = value
+        self._dao.invoice_id = value
+
+    @property
+    def description(self):
+        """Return the Non-Sufficient Funds description."""
+        return self._description
+
+    @description.setter
+    def description(self, value: str):
+        """Set the Non-Sufficient Funds description."""
+        self._description = value
+        self._dao.description = value
+
+    def commit(self):
+        """Save the information to the DB."""
+        return self._dao.commit()
+
+    def rollback(self):
+        """Rollback."""
+        return self._dao.rollback()
+
+    def flush(self):
+        """Save the information to the DB."""
+        return self._dao.flush()
+
+    def save(self):
+        """Save the information to the DB."""
+        return self._dao.save()
+
+    def asdict(self):
+        """Return the Non-Sufficient Funds as a python dict."""
+        non_sufficient_funds_schema = NonSufficientFundsSchema()
+        d = non_sufficient_funds_schema.dump(self._dao)
+        return d
+
+    @staticmethod
+    def populate(value: NonSufficientFundsModel):
+        """Populate Non-Sufficient Funds Service."""
+        non_sufficient_funds_service: NonSufficientFundsService = NonSufficientFundsService()
+        non_sufficient_funds_service._dao = value  # pylint: disable=protected-access
+        return non_sufficient_funds_service
+
+    @staticmethod
+    def query_all_non_sufficient_funds_invoices(account_id: str, page: int, limit: int):
+        """Return all Non-Sufficient Funds invoices."""
+        query = db.session.query(PaymentModel, InvoiceModel) \
+            .join(PaymentAccountModel, PaymentAccountModel.id == PaymentModel.payment_account_id) \
+            .outerjoin(InvoiceReferenceModel, InvoiceReferenceModel.invoice_number == PaymentModel.invoice_number) \
+            .outerjoin(InvoiceModel, InvoiceReferenceModel.invoice_id == InvoiceModel.id) \
+            .join(NonSufficientFundsModel, InvoiceModel.id == NonSufficientFundsModel.invoice_id) \
+            .filter(PaymentAccountModel.auth_account_id == account_id) \
+            .filter(PaymentModel.paid_amount > 0)
+
+        query = query.order_by(PaymentModel.id.asc())
+        pagination = query.paginate(per_page=limit, page=page)
+        results, total = pagination.items, pagination.total
+
+        return results, total
+
+    @staticmethod
+    def find_all_non_sufficient_funds_invoices(account_id: str, page: int, limit: int):
+        # pylint: disable=too-many-locals
```

Comment:
We shouldn't be ignoring linting rules, if there are too many locals, probably need a refactor